CMakeLists.txt

```diff
@@ -2,11 +2,27 @@
 # Copyright (C) Electronic Arts Inc.  All rights reserved.
 #-------------------------------------------------------------------------------------------
 cmake_minimum_required(VERSION 3.1)
+project(EASTL CXX)
+
+#-------------------------------------------------------------------------------------------
+# Sub-projects
+#-------------------------------------------------------------------------------------------
+add_subdirectory(test/packages/EAAssert)
+add_subdirectory(test/packages/EABase)
+add_subdirectory(test/packages/EAStdC)
+add_subdirectory(test/packages/EAMain)
+add_subdirectory(test/packages/EATest)
+add_subdirectory(test/packages/EAThread)
+
+add_subdirectory(benchmark)
```

Comment:
If you add the cmake options you point out above I'll merge the PR.

>  This could be put inside CMake options like EASTL_BUILD_TESTS and EASTL_BUILD_BENCHMARK and only be referenced when these options are enabled.

Ensure you update the CI scripts and the contributing docs .  Thank you.